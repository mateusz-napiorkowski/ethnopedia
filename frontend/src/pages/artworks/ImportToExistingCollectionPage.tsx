import { ChangeEvent, useState } from "react";
import { useNavigate, useParams } from "react-router-dom";
import * as XLSX from 'xlsx';
import Navbar from "../../components/navbar/Navbar"
import Navigation from "../../components/Navigation";
import { ReactComponent as DragAndDrop } from "../../assets/icons/dragAndDrop.svg"
import { ReactComponent as ExcelIcon } from "../../assets/icons/excel.svg"
import { ReactComponent as CSVIcon } from "../../assets/icons/csv.svg"
import { ReactComponent as Close } from "../../assets/icons/close.svg";
import { ReactComponent as UnknownFile } from "../../assets/icons/unknown-file.svg";
import { useMutation, useQuery, useQueryClient } from "react-query";
import { importData } from "../../api/dataImport";
import { useUser } from "../../providers/UserProvider";
import { getAllCategories } from "../../api/categories";

const ImportToExistingCollectionPage = () => {
    const params = useParams()
    const collectionId = params.collection
    const nbsp = "\u00A0"
    const [fileLoaded, setFileLoaded] = useState(false)
    const [fileName, setFileName] = useState<string>("")
    const [fileData, setFileData]: any = useState(false)
    const [excelCollectionCategoryPairs, setExcelCollectionCategoryPairs]: any = useState([])
    const [fileNotLoadedError, setFileNotLoadedError] = useState(nbsp)
    const [serverError, setServerError] = useState(nbsp)
    let dataToSend: string[][] = []
    
    const navigate = useNavigate();
    const queryClient = useQueryClient()
    const { jwtToken } = useUser();
    
    const { data: categoriesData } = useQuery({
        queryKey: ["allCategories"],
        queryFn: () => getAllCategories([collectionId!]),
        enabled: !!collectionId,
    })

    const removeEmptyColumns = (data: string[][]): string[][] => {
        if (data.length === 0) return data;

        const maxColumns = Math.max(...data.map(row => row.length));

        const columnsToKeep = Array.from({ length: maxColumns }, (_, i) =>
            data.some(row => row[i] !== undefined && row[i] !== "")
        );

        return data.map(row =>
            row.filter((_, i) => columnsToKeep[i])
        );
    };

    const handleFileUpload = (event: any) => {
        const file = event.target.files?.[0]
        if(!file) return

        const reader = new FileReader();
        reader.onload = (evt: any) => {
            const bString = evt.target.result;
            const workbook = XLSX.read(bString, {type:'binary'});

            const worksheetName = workbook.SheetNames[0];
            const worksheet = workbook.Sheets[worksheetName];

            const parsedData: Array<Array<string>> = removeEmptyColumns(
                XLSX.utils.sheet_to_json(worksheet, {header:1, defval: "", raw: false})
            );

            setFileLoaded(true)
            setFileName(file.name)
            setFileData(parsedData)
            setExcelCollectionCategoryPairs(parsedData[0]
                .map((headerCategory: string) => {
                    if(headerCategory === "_id")
                        return [headerCategory, "_id"]
                    if(headerCategory === "nazwy plików")
                        return [headerCategory, "nazwy plików"]
                    const matchedCollectionCategory = categoriesData.categories.find((collectionCategory: string) => collectionCategory.toLowerCase() === headerCategory.toLowerCase())
                    return [headerCategory, matchedCollectionCategory]
                })
            )
            setFileNotLoadedError(nbsp)
        };
        reader.readAsArrayBuffer(file)
        event.target.value = "";
    }

    const handleFileRemove = (event: any) => {
        event.preventDefault()
        setFileLoaded(false)
        setFileName("")
        setFileData(false)
        setExcelCollectionCategoryPairs([])
    };

    const handleOptionChange = ((event: ChangeEvent<HTMLSelectElement>) => {
        const fileCategory = event.target.id.replace(/-collection-equivalent$/, "");
        const collectionCategory = event.target.value
        setExcelCollectionCategoryPairs((prevPairs: Array<Array<string>>) =>
            prevPairs.map(([prevFileCategory, prevCollectionCategory]) =>
                prevFileCategory === fileCategory ? [prevFileCategory, collectionCategory] : [prevFileCategory, prevCollectionCategory]
            )
        );
    })

    const showServerError = ((error: any) => {
        if(error.error == 'Incorrect request body provided')
            setServerError("Import kolekcji nie powiódł się z powodu nieprawidłowej treści żądania. Upewnij się, że plik arkusza kalkulacyjnego zawiera przynajmniej jeden rekord oprócz nagłówka.")
        else if(error.error == "Invalid data in the spreadsheet file")
            setServerError("Nieprawidłowe dane w pliku arkusza kalkulacyjnego. Upewnij się, że kategorie zostały poprawnie wczytane, i że powyższy formularz został wypełniony prawidłowo.")
        else if(error.error == `Collection not found`)
            setServerError("Nie znaleziono kolekcji, do której dane miały zostać wprowadzone.")
        else
            setServerError("Błąd serwera. Import kolekcji nie powiódł się.")
    })
    
    const handleCollectionSubmit = (event: any) => {
        event.preventDefault()
        setFileNotLoadedError(fileLoaded ? nbsp : "Nie wczytano pliku")
        const newHeader = excelCollectionCategoryPairs.map((pair: String[]) => pair[1])
        dataToSend = [newHeader, ...fileData.slice(1)]
        importDataMutation.mutate()
    }

    const importDataMutation = useMutation(() => importData(dataToSend, jwtToken, collectionId!), {
        onSuccess: () => {
            queryClient.invalidateQueries("collection")
            navigate(`/collections/${collectionId}/artworks`)
        },
        onError: (error: any) => {
            showServerError(error.response.data)
        }
    })

    const FileExtensionIcon: React.FC<{name: string;}> = ({name}) => {
        if((/\.(csv|tsv|txt)$/i.test(name)))
            return (<CSVIcon className="w-12 h-12"/>)
        else if((/\.(xlsx|xls|xlsm|xlsb|ods|xltx|xltm)$/i.test(name)))
            return (<ExcelIcon className="w-12 h-12"/>)
        else
            return (<UnknownFile className="w-12 h-12"/>)
    }

    return (
        <div className="min-h-screen bg-gray-50 dark:bg-gray-900 flex flex-col">
            <Navbar />
            <div className="container mx-auto px-24 sm:px-32 md:px-40 lg:px-48 mt-4 max-w-screen-lg">
                <Navigation />
                <div className="mt-4">
                    <div className="bg-white dark:bg-gray-800 rounded-lg shadow-md border dark:border-gray-600 p-8">
                        <div className="flex items-start rounded-t border-b pb-2">
                            <h3 className="text-2xl font-bold text-gray-900 dark:text-white">
                                Importuj dane do kolekcji
                            </h3>
                        </div>
                        <form onSubmit={handleCollectionSubmit}>
                            <label
                                htmlFor="dropzone-file"
                                className="block text-sm font-bold text-gray-700 dark:text-white my-2"
                            >
                                Wgraj plik arkusza kalkulacyjnego/CSV
                            </label>
                            <label
                                aria-label="upload"
                                htmlFor="dropzone-file"
                                className="flex flex-col items-start justify-start p-2 border-2 border-gray-200
                                            border-dashed rounded-lg cursor-pointer bg-gray-50 dark:hover:bg-gray-600
                                            dark:bg-gray-800 hover:bg-gray-100 dark:border-gray-600
                                            dark:hover:border-gray-500 dark:hover:bg-gray-700"
                            >
                                {fileLoaded 
                                    ? <div className="flex flex-row items-center justify-between w-full">
                                        <div className="flex flex-row items-center justify-center gap-4">
                                            <FileExtensionIcon name={fileName} />
                                            <p className="text-sm font-bold text-gray-500 dark:text-gray-400">
                                                {fileName}
                                            </p>          
                                        </div>
                                        <button
                                            aria-label="remove-file-to-load"
                                            type="button"
                                            className="text-gray-400 hover:bg-gray-200 hover:text-gray-900 text-sm
                                                    dark:hover:bg-gray-600 dark:hover:text-white p-2 rounded-lg cursor-pointer"
                                            onClick={handleFileRemove}
                                        >
                                            <Close />
                                        </button>
                                      </div>
                                    : <div className="flex flex-row items-center justify-center gap-4">
                                        <DragAndDrop className="w-12 h-12 text-gray-500 dark:text-gray-400"/>
                                        <p className="text-sm font-normal text-gray-500 dark:text-gray-400">
                                            Kliknij, aby przesłać plik
                                        </p>
                                    </div>
                                }
                                <input
                                    id="dropzone-file"
                                    accept=".xlsx,.xls,.xlsm,.xlsb,.ods,.csv,.tsv,.txt,.xltx,.xltm"
                                    type="file"
                                    className="hidden"
                                    onChange={handleFileUpload}
                                />
                            </label>
                            <p
                                className={`block text-sm ${fileNotLoadedError != nbsp ? "text-red-500 font-normal": "font-semibold text-gray-700 dark:text-white"} my-2`}
                            >
                                {fileLoaded && (/\.(csv|tsv|txt|xlsx|xls|xlsm|xlsb|ods|xltx|xltm)$/i.test(fileName)) && (
                                    <span>
                                        Liczba rekordów: <span className="font-normal">{fileData.length - 1}</span>
                                    </span>
                                )}
                            </p>

                            <hr />
                            {fileLoaded && ( <>
                                <p className="text-sm font-normal text-gray-700 dark:text-white my-2">W arkuszu wykryto kategorie widoczne w lewej kolumnie poniżej. Upewnij się czy odpowiadają one odpowiednim kategoriom kolekcji, do której dane mają zostać zaimportowane. Jeśli nie, to wprowadź poniżej odpowienie poprawki.</p>
                                <div className="flex flex-col items-center justify-center mt-4 mb-4 p-4 border-2 border-gray-200
                                    border-solid rounded-lg bg-gray-50 dark:hover:bg-gray-600
                                    dark:bg-gray-800 dark:border-gray-600
                                    dark:hover:border-gray-500">
                                    <div className="flex flex-row w-full items-start justify-start">
                                        <span className="block w-1/2 text-sm font-semibold text-gray-700 dark:text-white my-2">Kategorie wgranego pliku:</span>
                                        <span className="block w-1/2 text-sm font-semibold text-gray-700 dark:text-white my-2">Kategorie w kolekcji:</span>
                                    </div>
                                    {excelCollectionCategoryPairs
                                        .filter((pair: string) => pair[0] !== "_id" && pair[0] !== "nazwy plików")
                                        .map((pair: any) => {
                                            const headerCategoryName = pair[0]
                                            const collectionCategoryName = pair[1]
                                            return (
                                                <div className="flex flex-row w-full items-start justify-start">
                                                    <label
                                                        htmlFor={`${headerCategoryName}-collection-equivalent`}
                                                        className="block w-1/2 text-sm font-semibold text-gray-700 dark:text-white my-2"
                                                    >
                                                        {headerCategoryName}
                                                    </label>
                                                    <select 
                                                        className="block w-1/2 text-sm font-semibold text-gray-700 dark:text-white my-2"
                                                        id={`${headerCategoryName}-collection-equivalent`}
                                                        aria-label={`${headerCategoryName}-collection-equivalent-select`}
                                                        onChange={handleOptionChange}   
                                                    >
                                                        {categoriesData.categories.map((optionValue: any) => {
                                                            return (<option
                                                                selected={optionValue == collectionCategoryName ? true : false}
                                                                value={optionValue}
                                                                >
                                                                    {optionValue}
                                                                </option>)
                                                        })}                                                            
                                                    </select>
                                                </div>
                                            )
                                        })
                                    }
                                </div>
                            </>)}
                            <div 
                                aria-label="server-error"
                                className="text-red-500 text-sm"
                            >
                                {serverError}
                            </div>
                            <div className="flex justify-end mt-6">
                                <button
                                    type="button"
                                    onClick={() => navigate(-1)}
                                    className="px-4 py-2 mr-2"
                                >
                                    Anuluj
                                </button>
                                <button
                                    aria-label="import-data"
                                    type="submit"
                                    disabled={!fileLoaded ? true : false }
                                    className="px-4 py-2 color-button"
                                >
                                    Importuj dane
                                </button>
                            </div>
                        </form>
                    </div>
                </div>
            </div>
        </div>
    );
}

export default ImportToExistingCollectionPage