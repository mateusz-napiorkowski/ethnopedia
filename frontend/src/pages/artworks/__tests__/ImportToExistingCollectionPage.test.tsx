import '@testing-library/jest-dom';
import { render } from '@testing-library/react'
import ImportToExistingCollectionPage from "../ImportToExistingCollectionPage"
import { MemoryRouter, Routes, Route } from "react-router-dom";
import userEvent from "@testing-library/user-event";
import { QueryClient, QueryClientProvider } from "react-query";
// import { Collection } from "../../@types/Collection"
// import { UserContext } from '../../providers/UserProvider';

const mockImportData = jest.fn()
jest.mock('../../../api/dataImport', () => ({
    importData: (
        importData: string[][],
        jwtToken: string,
        collectionId: string
    ) => mockImportData(importData, jwtToken, collectionId),
}))

const queryClient = new QueryClient();
const user = userEvent.setup()

const exampleCollectionData = {
    _id: '662e928b11674920c8cc0aaa',
    name: 'example collection',
    description: 'example collection description'
}
const jwtToken = "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJ1c2VybmFtZSI6InRlc3Rvd3kiLCJmaXJzdE5hbWUiOiJ0ZXN0b3d5IiwidXN"
    + "lcklkIjoiNjZiNjUwNmZiYjY0ZGYxNjVlOGE5Y2U2IiwiaWF0IjoxNzI0MTg0MTE0LCJleHAiOjE3MjUxODQxMTR9.fzHPaXFMzQTVUf9IdZ0G6oeiaecc"
    + "N-rDSjRS3kApqlA"

const exampleCollectionId = "67f84d80d2ac8e9a1e67cca4"
const renderComponent = (collectionId = exampleCollectionId) => {
    return render(
        <QueryClientProvider client={queryClient}>
            <MemoryRouter initialEntries={[`/collections/${collectionId}/import-data/`]}>
                <Routes>
                    <Route path="/collections/:collection/import-data" element={<ImportToExistingCollectionPage />}/>
                </Routes>  
            </MemoryRouter>
        </QueryClientProvider>
        
    );
};

describe("ImportToExistingCollectionPage tests", () => {
    beforeEach(() => {
        queryClient.clear();
        jest.clearAllMocks()
    });

    it("should render initial state", () => {           
        const {container} = renderComponent()

        expect(container).toMatchSnapshot()
    })

    // it("should render menu for importing data to existing collection, import metadata button should be disabled", () => {           
    //     const {getByText} = renderComponent("/collections/example collection/artworks", exampleCollectionData)
    //     const importMetadataButton = getByText(/importuj metadane/i)

    //     expect(getByText(/kliknij, aby przesłać/i)).toBeInTheDocument()
    //     expect(importMetadataButton).toBeDisabled()
    // })

    // it("should call onClose when exit button is clicked", async () => {           
    //     const { getByLabelText } = renderComponent("/collections/example collection/artworks", exampleCollectionData)
    //     const exitButton = getByLabelText("exit")

    //     await user.click(exitButton)

    //     expect(mockOnClose).toHaveBeenCalled()
    // })

    // it("should upload file, file name should be shown and next button should be enabled", async () => {           
    //     const {getByText, getByLabelText} = renderComponent()
    //     const nextButton = getByText(/dalej/i)
    //     const file = new File(['example'], 'example.xlsx', {type: 'application/vnd.openxmlformats-officedocument.spreadsheetml.sheet'})
    //     const input = getByLabelText("upload")

    //     await user.upload(input, file)

    //     expect(getByText("example.xlsx")).toBeInTheDocument()
    //     expect(nextButton).toBeEnabled()
    // })

    // it("should go to collection details menu", async () => {           
    //     const {getByText, getByLabelText} = renderComponent()
    //     const nextButton = getByText("Dalej")
        
    //     const file = new File(['example'], 'example.xlsx', {type: 'application/vnd.openxmlformats-officedocument.spreadsheetml.sheet'})
    //     const input = getByLabelText("upload")

    //     await user.upload(input, file)
    //     await user.click(nextButton)

    //     expect(getByText(/nazwa kolekcji/i)).toBeInTheDocument()
    //     expect(getByText(/opis kolekcji/i)).toBeInTheDocument()        
    // })

    // it.each([
    //     {
    //         case: "disabled when nothing was typed in",
    //         actions: [],
    //         enabled: false
    //     },
    //     {
    //         case: "disabled when only collection name is typed in",
    //         actions: [
    //             {
    //                 textAreaLabelText: "name",
    //                 textToTypeIn: "example collection name"
    //             }
    //         ],
    //         enabled: false
    //     },
    //     {
    //         case: "disabled when only collection description is typed in",
    //         actions: [
    //             {
    //                 textAreaLabelText: "description",
    //                 textToTypeIn: "example description name"
    //             }
    //         ],
    //         enabled: false
    //     },
    //     {
    //         case: "enabled when both collection name and description are typed in",
    //         actions: [
    //             {
    //                 textAreaLabelText: "name",
    //                 textToTypeIn: "example collection name"
    //             },
    //             {
    //                 textAreaLabelText: "description",
    //                 textToTypeIn: "example description name"
    //             }
    //         ],
    //         enabled: true
    //     },
    // ])('should have import metadata button $case in collection details menu', async ({actions, enabled}) => {
    //     const {getByText, getByLabelText} = renderComponent()
    //     const nextButton = getByText(/dalej/i)
    //     const file = new File(['example'], 'example.xlsx', {type: 'application/vnd.openxmlformats-officedocument.spreadsheetml.sheet'})
    //     const input = getByLabelText("upload")
    //     await user.upload(input, file)
    //     await user.click(nextButton)
    //     const importMetadataButton = getByText(/importuj metadane/i)

    //     for(const {textAreaLabelText, textToTypeIn} of actions) {
    //         const textArea = getByLabelText(textAreaLabelText)
    //         if(textToTypeIn) {
    //             await user.type(textArea, textToTypeIn)
    //         } else {
    //             await user.clear(textArea)
    //         }
    //     }
        
    //     if(enabled) {
    //         expect(importMetadataButton).toBeEnabled()
    //     } else {
    //         expect(importMetadataButton).toBeDisabled()
    //     }
    // })

    // it("should go to back to file upload menu when back button is clicked, switching between upload/collection menus should not reset their state", async () => {           
    //     const {getByText, getByLabelText, queryByText} = renderComponent()
    //     const nextButton = getByText(/dalej/i)
    //     const file = new File(['example'], 'example.xlsx', {type: 'application/vnd.openxmlformats-officedocument.spreadsheetml.sheet'})
    //     const input = getByLabelText("upload")

    //     await user.upload(input, file)
    //     await user.click(nextButton)
    //     const backButton = getByText(/wstecz/i)
    //     const importMetadataButton = getByText(/importuj metadane/i)
    //     const nameTextArea = getByLabelText("name")
    //     const descriptionTextArea = getByLabelText("description")

    //     await user.type(nameTextArea, "example collection")
    //     await user.type(descriptionTextArea, "example collection description")

    //     await user.click(backButton)
    //     expect(queryByText("example.xlsx")).toBeInTheDocument()
    //     expect(queryByText("example collection")).not.toBeInTheDocument()
    //     expect(queryByText("example collection description")).not.toBeInTheDocument()

    //     await user.click(getByText("Dalej"))
    //     expect(queryByText("example.xlsx")).not.toBeInTheDocument()
    //     expect(queryByText("example collection")).toBeInTheDocument()
    //     expect(queryByText("example collection description")).toBeInTheDocument()

    //     expect(importMetadataButton).toBeEnabled()
    // })

    // it("should call mockImportData with correct args after file is uploaded and then import metadata button is clicked", async () => {
    //     const fileBits = ['example']
    //     const {getByText, getByLabelText, queryByText} = renderComponent("/collections/example collection/artworks", exampleCollectionData)
    //     const file = new File(fileBits, 'example.xlsx', {type: 'application/vnd.openxmlformats-officedocument.spreadsheetml.sheet'})
    //     const input = getByLabelText("upload")
    //     const importMetadataButton = getByText(/importuj metadane/i)

    //     await user.upload(input, file)
    //     await user.click(importMetadataButton)

    //     expect(mockImportData).toHaveBeenCalledWith([fileBits], jwtToken, exampleCollectionData._id)
    // })

    // it("should call mockImportDataAsCollection with correct parameters after file, name, and description are provided by user and import metadata button is clicked", async () => {
    //     const {getByText, getByLabelText, queryByText, container} = renderComponent()
    //     const nextButton = getByText(/dalej/i)
    //     const fileBits = ['example']
    //     const file = new File(fileBits, 'example.xlsx', {type: 'application/vnd.openxmlformats-officedocument.spreadsheetml.sheet'})
    //     const input = getByLabelText("upload")

    //     await user.upload(input, file)
    //     await user.click(nextButton)

    //     const importMetadataButton = getByText(/importuj metadane/i)
    //     const nameTextArea = getByLabelText("name")
    //     const descriptionTextArea = getByLabelText("description")

    //     await user.type(nameTextArea, "example collection")
    //     await user.type(descriptionTextArea, "example collection description")
    //     await user.click(importMetadataButton)

    //     expect(mockImportDataAsCollection).toHaveBeenCalledWith([fileBits], "example collection", "example collection description", jwtToken)
    // })
})