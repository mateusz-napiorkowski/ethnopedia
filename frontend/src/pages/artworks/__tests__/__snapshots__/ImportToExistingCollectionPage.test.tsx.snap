// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImportToExistingCollectionPage tests should render initial state 1`] = `
<div>
  <div
    class="min-h-screen bg-gray-50 dark:bg-gray-900 flex flex-col"
  >
    <nav
      class="bg-gray-800 border-gray-200 dark:bg-gray-200 w-full border-b dark:border-gray-600"
    >
      <div
        class="max-w-screen-xl flex flex-wrap justify-between mx-auto p-1"
      >
        <div
          class="flex items-center mt-1"
        >
          <h1
            class="logo text-gray-200 dark:text-gray-900 text-6xl "
          >
            ethnopedia
          </h1>
        </div>
        <div
          class="flex items-center md:order-2"
        >
          <button
            class="text-gray-800 bg-white border border-gray-300 focus:outline-none
                           hover:bg-gray-100 focus:ring-4 focus:ring-gray-200 font-medium rounded-lg text-sm
                           p-2.5 mr-2 dark:bg-gray-800 dark:text-white dark:border-gray-600
                           dark:hover:bg-gray-700 dark:hover:border-gray-600 dark:focus:ring-gray-700"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="w-5 h-5 text-gray-800 dark:text-gray-200"
              fill="none"
              height="1em"
              stroke="currentColor"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M21 21l-6-6m2-5a7 7 0 11-14 0 7 7 0 0114 0z"
                stroke-linecap="round"
                stroke-linejoin="round"
              />
            </svg>
          </button>
          <button
            class="text-gray-800 bg-white border border-gray-300 focus:outline-none
                           hover:bg-gray-100 focus:ring-4 focus:ring-gray-200 font-medium rounded-lg text-sm
                           p-2.5 mr-2 dark:bg-gray-800 dark:text-white dark:border-gray-600
                           dark:hover:bg-gray-700 dark:hover:border-gray-600 dark:focus:ring-gray-700"
            type="button"
          >
            <svg
              class="w-4 h-4 m-0.5"
            />
          </button>
          <button
            class="mr-2 bg-blue-500 hover:bg-blue-400 font-semibold text-white border-none"
            type="button"
          >
            Zaloguj się
          </button>
          <button
            class="bg-blue-500 hover:bg-blue-400 font-semibold text-white border-none"
            type="button"
          >
            Zarejestruj się
          </button>
        </div>
      </div>
    </nav>
    <div
      class="container mx-auto px-24 sm:px-32 md:px-40 lg:px-48 mt-4 max-w-screen-lg"
    >
      <nav
        class="flex"
      >
        <ol
          class="inline-flex items-center space-x-1 md:space-x-2"
        >
          <li
            class="inline-flex items-center text-lg"
          >
            <a
              class="inline-flex items-center font-medium text-gray-600 dark:text-gray-400 hover:text-blue-600 dark:hover:text-blue-600"
              href="/"
            >
              Strona główna
            </a>
          </li>
          <span
            class="self-center"
          >
            <svg
              class="w-3 h-3 text-gray-600 dark:text-gray-400"
            />
          </span>
          <li
            class="inline-flex items-center text-lg"
          >
            <a
              class="inline-flex items-center font-medium text-gray-600 dark:text-gray-400 hover:text-blue-600 dark:hover:text-blue-600"
              href="/collections/67f84d80d2ac8e9a1e67cca4/artworks"
            >
              Kolekcja
            </a>
          </li>
        </ol>
      </nav>
      <div
        class="mt-4"
      >
        <div
          class="bg-white dark:bg-gray-800 rounded-lg shadow-md border dark:border-gray-600 p-8"
        >
          <div
            class="flex items-start rounded-t border-b pb-2"
          >
            <h3
              class="text-2xl font-bold text-gray-900 dark:text-white"
            >
              Importuj dane do kolekcji
            </h3>
          </div>
          <form>
            <label
              class="block text-sm font-bold text-gray-700 dark:text-white my-2"
              for="dropzone-file"
            >
              Wgraj plik arkusza kalkulacyjnego/CSV
            </label>
            <label
              aria-label="upload"
              class="flex flex-col items-start justify-start p-2 border-2 border-gray-200
                                            border-dashed rounded-lg cursor-pointer bg-gray-50 dark:hover:bg-gray-600
                                            dark:bg-gray-800 hover:bg-gray-100 dark:border-gray-600
                                            dark:hover:border-gray-500 dark:hover:bg-gray-700"
              for="dropzone-file"
            >
              <div
                class="flex flex-row items-center justify-center gap-4"
              >
                <svg
                  class="w-12 h-12 text-gray-500 dark:text-gray-400"
                />
                <p
                  class="text-sm font-normal text-gray-500 dark:text-gray-400"
                >
                  Kliknij, aby przesłać plik
                </p>
              </div>
              <input
                class="hidden"
                id="dropzone-file"
                type="file"
              />
            </label>
            <p
              class="block text-sm font-semibold text-gray-700 dark:text-white my-2"
            />
            <hr />
            <div
              class="flex justify-end mt-6"
            >
              <button
                class="px-4 py-2 mr-2"
                type="button"
              >
                Anuluj
              </button>
              <button
                class="px-4 py-2 color-button"
                disabled=""
                type="submit"
              >
                Importuj dane
              </button>
            </div>
          </form>
        </div>
      </div>
    </div>
  </div>
</div>
`;
